src/praisonai/praisonai/bots/_media.py

```
@@ -0,0 +1,242 @@
+        ):
+            return False
+    return True
+
+
+def _fetch_url(url: str, max_bytes: int) -> bytes:
+    """Fetch ``url`` with an SSRF guard and hard size cap. Returns bytes."""
+    if not _is_safe_url(url):
+        raise InboundMediaError(f"Refusing to fetch unsafe inbound media URL: {url!r}")
+    import urllib.request
+
+    req = urllib.request.Request(url, headers={"User-Agent": "PraisonAI-Bot"})
```

Ruff (0.15.18)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)